### PR TITLE
Updates and fixes according to the SignalR RP changes

### PR DESCRIPTION
**Pull request by allxiao**
## Description

* Fix positional attributes for mandatory parameters
* Use the default resource group set by `Set-AzureRmDefault` when we call `New-AzureRmSignalR`, if not specified. And fallback to use the `Name` if the default is not set either.
* Remove the lowercase restriction to the `Name` parameter
* Add `UnitCount` parameter for the `New-AzureRmSignalR`, to specify the capacity in SKU. Help docs was updated accordingly.
* Add a new SKU option `Free_DS2`.

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
    - [x] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: src/ResourceManager/SignalR/Commands.SignalR/Cmdlets/GetAzureRmSignalR.cs [hunk 1]
```diff
@@ -25,7 +25,7 @@ namespace Microsoft.Azure.Commands.SignalR.Cmdlets
     [OutputType(typeof(PSSignalRResource))]
     public class GetAzureRmSignalR : SignalRCmdletBase, IWithResourceId
     {
-        [Parameter(
+        [Parameter(Position = 0,
```
Comment (sergey-shandar) [inline src/ResourceManager/SignalR/Commands.SignalR/Cmdlets/GetAzureRmSignalR.cs]: I don't think we should use Position in a non-mandatory parameter.

Comment (ArieShout) [reply src/ResourceManager/SignalR/Commands.SignalR/Cmdlets/GetAzureRmSignalR.cs]: reverted and updated the position for other mandatory parameters after discussion with Sergey. Docs were updated accordingly.

**Comment by sergey-shandar** [comment]
Signed build https://azuresdkci.westus2.cloudapp.azure.com/job/ps-sign/562/

**Comment by markcowl** [comment]
on demand: https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/powershell-demand/804/

**Pull request merged**